src/rgw/rgw_rest_s3.cc

```diff
@@ -2120,6 +2129,9 @@ RGWOp *RGWHandler_ObjStore_Obj_S3::op_put()
   if (is_acl_op()) {
     return new RGWPutACLs_ObjStore_S3;
   }
+  if (is_rename_op() && store->ctx()->_conf->rgw_enable_rename_op) {
```

Comment:
I think this function will return to RGWCopyObj_ObjStore_S3 in case enable_rename_op is disabled, is this the right behaviour?